### PR TITLE
Fix HCAL TP compression LUTs by populating plan 1 TP list first.

### PR DESCRIPTION
**Pull request by matz-e**
Backport of #17789.

### CODE REVIEW

**Comment by franzoni** [comment]
please test

File: CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc [hunk 1]
```diff
@@ -253,21 +253,21 @@ void CaloTPGTranscoderULUT::setup(HcalLutMetadata const& lutMetadata, HcalTrigTo
...
+    plan1_towers_.clear();
+    for (const auto& id: lutMetadata.getAllChannels()) {
+       if (not (id.det() == DetId::Hcal and theTopology->valid(id)))
+          continue;
+       HcalDetId cell(id);
+       if (not theTopology->dddConstants()->isPlan1(cell))
+          continue;
+       for (const auto& tower: theTrigTowerGeometry.towerIds(cell))
+          plan1_towers_.emplace(tower);
+    }
```
Comment (franzoni) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc]: hello @matz-e 
plan1_towers_ 
is filled only at this line, inside CaloTPGTranscoderULUT::getOutputLUTSize(...),
and loadHCALCompress() does not seem to affect its content.

Therefore,
why does the swapping the ordering matter ?

Comment (matz-e) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc]: `loadHCALCompress()` calls `getOutputLUTSize()` [here](https://github.com/matz-e/cmssw/blob/85afb830f94d11c470ab3fe74233aa18e0e521f8/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc#L74), and thus relies on `plan1_towers` already filled.

Comment (matz-e) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc]: `getOutputLUTSize()` uses `plan1_towers_` [in line 212](https://github.com/matz-e/cmssw/blob/85afb830f94d11c470ab3fe74233aa18e0e521f8/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc#L212), that's why the order matters…

**Comment by deguio** [comment]
@rekovic 
this is needed in 900. could you sign please?
thanks!
F.

**Comment by davidlange6** [comment]
thanks for the reminder @deguio - will merge

**Pull request merged**